src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py

```diff
@@ -526,14 +563,105 @@ def test_aic_lowertolanguagelevel_multi_invoke():
+    including where the nodes are ancestors of each other.
+
+    '''
+    code = (
+        "module mod1\n"
+        "contains\n"
+        "subroutine alg1()\n"
+        "  use kern_mod, only : kern1\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  call invoke(kern1(field))\n"
+        "end subroutine alg1\n"
+        "subroutine alg2()\n"
+        "  use kern_mod, only : kern2\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  call invoke(kern2(field))\n"
+        "end subroutine alg2\n"
+        "end module\n")
+
+    psyir = create_alg_psyir(code)
+    # Add an invoke symbol to the module (Container node)
+    container = psyir.children[0]
+    assert isinstance(container, Container)
+    container.symbol_table.add(RoutineSymbol("invoke"))
+
+    invoke1 = container.children[0][0]
+    assert isinstance(invoke1, AlgorithmInvokeCall)
+    invoke2 = container.children[1][0]
+    assert isinstance(invoke2, AlgorithmInvokeCall)
+
+    assert invoke1.scope.symbol_table.lookup(
+        "invoke", scope_limit=invoke1.scope)
+    assert invoke2.scope.symbol_table.lookup(
+        "invoke", scope_limit=invoke2.scope)
+
+    psyir.lower_to_language_level()
+
+    call1 = container.children[0][0]
+    call2 = container.children[1][0]
+    with pytest.raises(KeyError) as info:
+        assert call1.scope.symbol_table.lookup(
+            "invoke", scope_limit=call1.scope)
+    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+    with pytest.raises(KeyError) as info:
+        assert call2.scope.symbol_table.lookup(
+            "invoke", scope_limit=call2.scope)
+    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+
+    container = psyir.children[0]
+    assert container.symbol_table.lookup("invoke")
+
+
+def test_aic_lowertolanguagelevel_invoke_symbols_scope():
+    '''Check that the lower_to_language_level method removes the
+    appropriate invoke symbol when it is in a different scope to the
+    invoke call.
+
+    '''
+    code = (
+        "subroutine alg()\n"
+        "  use kern_mod, only : kern\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  call invoke(kern(field))\n"
+        "end subroutine alg\n")
+    psyir = create_alg_psyir(code)
+    # Move the invoke so the invoke symbol is in a different scope to
+    # the invoke call.
+    invoke = psyir.children[0][0]
+    assert isinstance(invoke, AlgorithmInvokeCall)
+    symbol = invoke.scope.symbol_table.new_symbol(
+        root_name="i", symbol_type=DataSymbol, datatype=INTEGER_TYPE)
+    loop = Loop.create(
+        symbol, Literal("0", INTEGER_TYPE), Literal("1", INTEGER_TYPE),
+        Literal("1", INTEGER_TYPE), [invoke.detach()])
+    psyir.children[0].children.append(loop)
+
+    psyir.lower_to_language_level()
```

Comment:
Isn't this missing the check that the 'invoke' symbol has now been removed?

Reply:
It is. I must have run out of steam.